src/workerd/api/sockets.c++

```diff
@@ -139,6 +139,8 @@ jsg::Ref<Socket> connectImplNoOutputLock(
 
   auto& ioContext = IoContext::current();
 
+  JSG_REQUIRE(!ioContext.isFiddle(), TypeError, "Use `wrangler dev` to test this feature.");
```

Comment:
I don't love prescribing wrangler dev here. It assumes the context is one where using 'wrangler dev' is valid. Instead, can we just say something to the affect of 'Socket API not supported in xxx mode'